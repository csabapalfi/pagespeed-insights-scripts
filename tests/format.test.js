const {formatRow, formatHeading} = require('../lib/format');

describe('format', () => {
  describe('results', () => {
    const result = {
      name: 'run 1',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };

    it('results are formatted as expected', () => {
      expect(formatRow(result)).toMatchSnapshot();
    });
  
    it('heading row is formatted as expected', () => {
      expect(formatHeading(result)).toMatchSnapshot();
    });
  });

  describe('stats', () => {
    const statistic = {
      name: 'median',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };
  
    it('stats are formatted as expected', () => {
      expect(formatRow(statistic))
        .toMatchSnapshot();
    });

    it('stddev uses custom precision for score (1 digit)', () => {
      expect(formatRow({...statistic, name: 'stddev'}))
        .toMatchSnapshot();
    });
  });
});