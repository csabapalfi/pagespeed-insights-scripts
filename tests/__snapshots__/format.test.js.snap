// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format results heading row is formatted as expected 1`] = `"name  	score	FCP	SI	LCP	TTI	TBT	CLS"`;

exports[`format results results are formatted as expected 1`] = `"run 1 	[32m95[39m	0.9	1.1	1.0	4.0	5.0	0.00"`;

exports[`format stats stats are formatted as expected 1`] = `"median	95	0.9	1.1	1.0	4.0	5.0	0.00"`;

exports[`format stats stddev uses custom precision for score (1 digit) 1`] = `"stddev	95.0	0.9	1.1	1.0	4.0	5.0	0.00"`;
